adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -677,24 +672,15 @@ private void doUploadMessage(
                                 contentType,
                                 payload,
                                 props,
-                                currentSpan.context()),
-                        responseReady.future())
+                                currentSpan.context()).onFailure(thr -> responseReadyTracker.cancel("send event failed")),
+                        responseReadyTracker.waitForCommand())
                         .map(s -> (Void) null);
             }
         })
-        .compose(proceed -> {
+        .onSuccess(proceed -> {
             // overwrites the execution of closeHandler in setTtdRequestConnectionCloseHandler
             // to prevent ctx.response().end() from being called twice
             ctx.response().closeHandler(v -> logResponseGettingClosedPrematurely(ctx.getRoutingContext()));
```

Comment:
This block here could be moved to the place where the `responseReadyTracker` future gets completed.

But this could be also done in a subsequent refactoring PR.

Reply:
The obvious step would be to move it to the following `map` method.
However, having in mind that response close in just in map/recover of the response ready (which was one of the aims of this PR) I don't think that this close handler switch is needed at all. 
So it shall be removed. WDYT?

Reply:
Yes, it doesn't seem needed anymore.